spyder/utils/ipython/style.py

```diff
@@ -0,0 +1,215 @@
+    string_font_style = give_font_style(str_fs)
+    num_c, num_fw, num_fs = color_scheme['number']
+    number_color =  num_c
+    number_font_weight = give_font_weight(num_fw)
+    number_font_style = give_font_style(num_fs)
+    com_c, com_fw, com_fs = color_scheme['comment']
+    comment_color =  com_c
+    comment_font_weight = give_font_weight(com_fw)
+    comment_font_style = give_font_style(com_fs)
+    def_c, def_fw, def_fs = color_scheme['definition']
+    definition_color =  def_c
+    definition_font_weight = give_font_weight(def_fw)
+    definition_font_style = give_font_style(def_fs)
+    ins_c, ins_fw, ins_fs = color_scheme['instance']
+    instance_color =  ins_c
+    instance_font_weight = give_font_weight(ins_fw)
+    instance_font_style = give_font_style(ins_fs)
+
+    syntax_style_dic = {}
+    syntax_style_dic['Name'] = [font_font_style, font_font_weight,
+                                font_color]
+    syntax_style_dic['Name.Class'] = [definition_font_style,
+                                      definition_font_weight,
+                                      definition_color]
+    syntax_style_dic['Name.Function'] = [definition_font_style,
+                                         definition_font_weight,
+                                         definition_color]
+    syntax_style_dic['Name.Builtin'] = [builtin_font_style,
+                                        builtin_font_weight, builtin_color]
+    syntax_style_dic['Name.Builtin.Pseudo'] = [instance_font_style,
+                                               instance_font_weight,
+                                               instance_color]
+    syntax_style_dic['Keyword'] = [keyword_font_style, keyword_font_weight,
+                                   keyword_color]
+    syntax_style_dic['Keyword.Type'] = [builtin_font_style,
+                                        builtin_font_weight, builtin_color]
+    syntax_style_dic['Comment'] = [comment_font_style, comment_font_weight,
+                                   comment_color]
+    syntax_style_dic['String'] = [string_font_style, string_font_weight,
+                                  string_color]
+    syntax_style_dic['Number'] = [number_font_style, number_font_weight,
+                                  number_color]
+    syntax_style_dic['Punctuation'] = [font_font_style, font_font_weight,
+                                       font_color]
+
+    return syntax_style_dic
+
+def get_syntax_style(name='spyder'):
+    """Create a .py with the spyder custom style as a Pygment class.
+
+    The file is stored in the pygment/styles folder.
+    Returns the name of this file (whithout path nor entension)
+    """
+
+    import pygments
+    syntax_path = os.path.join(pygments.__file__.rpartition(os.sep)[0],
+                               'styles')
+    syntax_name = name.replace('/', '').capitalize() + 'spyder'
+    syntax_filename = os.path.join(syntax_path, syntax_name + '.py')
```

Comment:
We can't do this! On Linux and some macOS package managers (Macports and Homebrew) we don't permissions to write in `pygment/styles`.

So we need to find a different approach @dalthviz. Looking at what `get_style_by_name` does (which is the reason for us to write this file)

```python
def get_style_by_name(name):
    if name in STYLE_MAP:
        mod, cls = STYLE_MAP[name].split('::')
        builtin = "yes"
    else:
        for found_name, style in find_plugin_styles():
            if name == found_name:
                return style
        # perhaps it got dropped into our styles package
        builtin = ""
        mod = name
        cls = name.title() + "Style"

    try:
        mod = __import__('pygments.styles.' + mod, None, None, [cls])
    except ImportError:
        raise ClassNotFound("Could not find style module %r" % mod +
                         (builtin and ", though it should be builtin") + ".")
    try:
        return getattr(mod, cls)
    except AttributeError:
        raise ClassNotFound("Could not find style class %r in style module." % cls)
```

Pygments is only trying to run a class that ends in `Style` from the modules present in `pygments.styles`. One of them is `VimStyle`, that looks like this:

```python
class VimStyle(Style):
    """
    Styles somewhat like vim 7.0
    """

    background_color = "#000000"
    highlight_color = "#222222"
    default_style = "#cccccc"

    styles = {
        Token:                     "#cccccc",
        Whitespace:                "",
        Comment:                   "#000080",
        Comment.Preproc:           "",
        Comment.Special:           "bold #cd0000",

        Keyword:                   "#cdcd00",
        Keyword.Declaration:       "#00cd00",
        Keyword.Namespace:         "#cd00cd",
        Keyword.Pseudo:            "",
        Keyword.Type:              "#00cd00",

        Operator:                  "#3399cc",
        Operator.Word:             "#cdcd00",

        Name:                      "",
        Name.Class:                "#00cdcd",
        Name.Builtin:              "#cd00cd",
        Name.Exception:            "bold #666699",
        Name.Variable:             "#00cdcd",

        String:                    "#cd0000",
        Number:                    "#cd00cd",

        Generic.Heading:           "bold #000080",
        Generic.Subheading:        "bold #800080",
        Generic.Deleted:           "#cd0000",
        Generic.Inserted:          "#00cd00",
        Generic.Error:             "#FF0000",
        Generic.Emph:              "italic",
        Generic.Strong:            "bold",
        Generic.Prompt:            "bold #000080",
        Generic.Output:            "#888",
        Generic.Traceback:         "#04D",

        Error:                     "border:#FF0000"
    }
```

So instead of using this function (and even the previous one, `get_syntax_style_dictionary`) we just need to define in this file a class called `SpyderStyle`, similar to `VimStyle` but that uses our color scheme.